### PR TITLE
New JSON output format!

### PR DESCRIPTION
**Pull request by TASelwyn**
The goal of this PR is to have a dynamic/decoupled telemetry output format so the UI doesn't need to understand what packet version and how they're decoded to be able to utilize it. A standard output no matter what packet versions the ground station receives. 

Also shuffles v1 data_block for classes to be in block number order, among a few other misc fixes.

### CODE REVIEW

**Comment by linguini1** [comment]
Is this almost ready for merge?

**Comment by TASelwyn** [comment]
> Is this almost ready for merge? No rush, just looking for a timeline.

All ready now. I just had one thing left to do.

File: modules/telemetry/json_packets.py
Comment (linguini1) [inline modules/telemetry/json_packets.py:208]: For future docstrings use the Sphinx format; that's what we're standardizing to.

File: modules/telemetry/json_packets.py
Comment (linguini1) [inline modules/telemetry/json_packets.py:201]: If you have a dataclass, don't define an init method.

File: modules/telemetry/json_packets.py
Comment (linguini1) [inline modules/telemetry/json_packets.py:213]: Are we sure this works? Won't `dict(self)` include method names?

File: config.json
Comment (linguini1) [inline config.json:3]: Just FYI, it's Red Bullistic

File: modules/telemetry/json_packets.py
Comment (linguini1) [inline modules/telemetry/json_packets.py:261]: Once again; data class should not have defined `__init__` methods. I would make this a regular class.

File: modules/telemetry/json_packets.py
Comment (linguini1) [inline modules/telemetry/json_packets.py:270]: Consider making this long type definition a dedicated type name:
```python
Specification: TypeAlias = dict[str, dict[str, dict[str, dict[str, str]]]]
```
(pick a better name that specification maybe idk)

File: modules/telemetry/json_packets.py
Comment (linguini1) [inline modules/telemetry/json_packets.py:293]: Snake case method names please! All my homies hate Java

File: modules/telemetry/json_packets.py
Comment (linguini1) [inline modules/telemetry/json_packets.py:332]: I think it might be beneficial to crash here. Without the schema we won't be able to output telemetry, no?

File: modules/telemetry/telemetry.py
Comment (linguini1) [inline modules/telemetry/telemetry.py:57]: You don't need to wrap it in `str()`, it's already passed as a string.

File: modules/telemetry/v1/data_block.py
Comment (linguini1) [inline modules/telemetry/v1/data_block.py:24]: No need to add reserved (even though it was there before). In the spec it just shows that anything > 0x08 isn't used yet.

**Review by linguini1 — CHANGES_REQUESTED** [review]
If you could (perhaps in a separate PR), also write some test cases for the new functionality you added.

File: modules/telemetry/v1/data_block.py
Comment (EliasJRH) [inline modules/telemetry/v1/data_block.py:222]: Nit: Add converter function from millidegrees to celsius since we have one for meters to feet

File: modules/telemetry/telemetry.py
Comment (EliasJRH) [inline modules/telemetry/telemetry.py:68]: Can we make the naming of this more explicit? `self.telemetry` is a confusing name when the class name is also `telemetry`. Maybe something like `self.data_buffer`?

File: modules/telemetry/json_packets.py
Comment (TASelwyn) [reply modules/telemetry/json_packets.py:213]: Before I override __iter__ it only returns  "mission_time" and all other stored variables. For example,

["mission_time", "celsius"]

Debug printing them on every update() returns this:
DEBUG:modules.telemetry.json_packets:UPDATING dict_keys(['mission_time', 'celsius']) using dict_keys(['celsius', 'mission_time'])!

File: modules/telemetry/json_packets.py
Comment (TASelwyn) [reply modules/telemetry/json_packets.py:332]: Upon testing this case only seems to gets thrown if the telemetry_packet.json and __iter__ outputs from that corresponding block in data_block gets thrown out of sync. I.e, if you put the metres variable as "altitude.met**re**s" but its actually "altitude.met**er**s" then it can't find the variable and will trigger. It's more of a sanity/verification when updating it. Additionally, I don't like the premise of crashing the whole software if a single variable is missing.

File: modules/telemetry/v1/data_block.py
Comment (TASelwyn) [reply modules/telemetry/v1/data_block.py:24]: Personally I agree with it being there for test cases/etc as a "default" value to set things to.

File: modules/telemetry/telemetry.py
Comment (TASelwyn) [reply modules/telemetry/telemetry.py:68]: self.telemetry_data 
👍

**Comment by TASelwyn** [comment]
Resolved all above comments. Please take a look again.

File: modules/misc/converter.py
Comment (linguini1) [inline modules/misc/converter.py:20]: No worries about adding the types in the docstring; those are apparent in the actual type annotations so omitting them will avoid duplication.

File: modules/telemetry/json_packets.py
Comment (linguini1) [reply modules/telemetry/json_packets.py:212]: Right, I was thinking of `self.__dict__`

**Review by linguini1 — CHANGES_REQUESTED** [review]
The only major thing for you to fix before this PR is merged is the `__init__` methods for the classes you removed the `@dataclass` decorators for. You still have dataclass style member declarations with `field()` which will create class attributes now that the classes are no longer data classes.

I left a few other comments that I'm hoping will be resolved at the same time, but as mentioned I won't block the PR over them.

File: modules/telemetry/json_packets.py
Comment (linguini1) [inline modules/telemetry/json_packets.py:257]: None of these serve a purpose, they're stub declarations for class attributes. I mean you need to delete all of these and only define them in `__init__` as instance attributes.

**Review by linguini1 — APPROVED** [review]
I'm approving this and I'll just remove the member attributes lol

File: modules/telemetry/json_packets.py
Comment (linguini1) [reply modules/telemetry/json_packets.py:257]: Resolved in 5180605.

**Pull request merged**
